### DISCUSSION TITLE
Dropping items on ground

### DISCUSSION THREAD

**Discussion by TreaBeardGaming** [peter-kish/gloot#200] [Q&A] [answered]
Before starting down the path of figuring out how to drop items on the ground as loot from various sources.... Does anyone have any ideas on which direction I should look to accomplish this?

I was thinking of using an area2d around the sources, instancing some nodes randomly in the area of the area2d, and then on player in range <= etc... add items to inventory and free the nodes used....

Is there a more elegant solution or  is it more a too each their own sorta thing on choosing how to execute

**Comment by peter-kish** [comment, marked as answer]
Unfortunately, this is another thing that is not fully supported by the plugin. In the future, I might implement some sort of item drop controls that can handle item drops.

But for now you can try the [dev_v2.4.8](https://github.com/peter-kish/gloot/tree/dev_v2.4.8) branch (hope to merge it soon), where you should be able to implement [`_drop_data`](https://docs.godotengine.org/en/stable/classes/class_control.html#class-control-private-method-drop-data) for any control and detect item drops (or any other kinds of UI control drops). Then it's just a matter of creating a control that is the size of the screen and placing it behind all other UI controls. Using the [`NOTIFICATION_DRAG_END`](https://docs.godotengine.org/en/stable/classes/class_node.html#constants) notification followed by a [`Viewport.gui_is_drag_successful()`](https://docs.godotengine.org/en/stable/classes/class_viewport.html#class-viewport-method-gui-is-drag-successful) check should also work.

And if you really want to use the master branch, you can create a transparent item slot that spans the entire screen and use the `ItemSlot.item_equipped` signal to detect the drop, immediately clear the slot and then handle the item however you like. Kinda ugly, but it should work :)

**Reply by TreaBeardGaming** [reply]
Oh thats an awesome idea with a control area for where to drop items.
I was thinking of just have an itemslot somewhere called dropme or something... but that would be a task for later.

In the meantime I just wrote some additional script for my resource nodes so when their health drops to 0 it will pull an item from the protoset of a specified type. Ofcourse there are some additional rules to randi for choosing what gets dropped from the item rate key.

I just made two additional inventory systems in my GUI that are way off screen, hidden, and can't be interacted with. But in short when a resource node is consumed it chooses an item, puts it into a temp inventory to set the stack size/etc then moves to an overflow inventory before going into the player inventory. I plan to have the overflow inventory transfer items at a set rate if there are slots available...

I'm using transfer_automerge() to move items from the overflow inventory into the player inventory. But when that returns false, I wanted to use transfer_autosplitmerge() to move what it can into slots that have room left in their stacks. However it returns "Assertion failed: Item count shouldn't be infinite".

submitted issue: #201 

![image](https://github.com/peter-kish/gloot/assets/67390743/745c0665-a384-48a4-a543-ff8f03c7df3d)
